File: frontend/src/app/model/validateSetup/validate-dampers.ts
export class ValidateDampers {
  frontBumpMin: number;
  frontBumpMax: number;
  frontFastBumpMin: number;
  frontFastBumpMax: number;
  frontReboundMin: number;
  frontReboundMax: number;
  frontFastReboundMin: number;
  frontFastReboundMax: number;
  rearBumpMin: number;
  rearBumpMax: number;
  rearFastBumpMin: number;
  rearFastBumpMax: number;
  rearReboundMin: number;
  rearReboundMax: number;
  rearFastReboundMin: number;
  rearFastReboundMax: number;

 ValidateFrontBump(frontLeftBump: number, frontRightBump: number) {
   return frontLeftBump >= this.frontBumpMin && frontLeftBump <= this.frontBumpMax
   && frontRightBump >= this.frontBumpMin && frontRightBump <= this.frontBumpMax;
}

ValidateFrontFastBump(frontLeftFastBump: number, frontRightFastBump: number) {
  return frontLeftFastBump >= this.frontFastBumpMin && frontLeftFastBump <= this.frontFastBumpMax
    && frontRightFastBump >= this.frontFastBumpMin && frontRightFastBump <= this.frontFastBumpMax;
}

ValidateFrontRebound(frontLeftRebound: number, frontRightRebound: number) {
  return frontLeftRebound >= this.frontReboundMin && frontLeftRebound <= this.frontReboundMax
  && frontRightRebound >= this.frontReboundMin && frontRightRebound <= this.frontReboundMax;
}

ValidateFrontFastRebound(frontLeftFastRebound: number, frontRightFastRebound: number) {
  return frontLeftFastRebound >= this.frontFastReboundMin && frontLeftFastRebound <= this.frontFastReboundMax
    && frontRightFastRebound >= this.frontFastReboundMin && frontRightFastRebound <= this.frontFastReboundMax;
}

ValidateRearBump(rearLeftBump: number, rearRightBump: number) {
  return rearLeftBump >= this.rearBumpMin && rearLeftBump <= this.rearBumpMax
  && rearRightBump >= this.rearBumpMin && rearRightBump <= this.rearBumpMax;
}

ValidateRearFastBump(rearLeftFastBump: number, rearRightFastBump: number) {
  return rearLeftFastBump >= this.rearFastBumpMin && rearLeftFastBump <= this.rearFastBumpMax
    && rearRightFastBump >= this.rearFastBumpMin && rearRightFastBump <= this.rearFastBumpMax;
}

ValidateRearRebound(rearLeftRebound: number, rearRightRebound: number) {
  return rearLeftRebound >= this.rearReboundMin && rearLeftRebound <= this.rearReboundMax
    && rearRightRebound >= this.rearReboundMin && rearRightRebound <= this.rearReboundMax;
}

ValidateRearFastRebound(rearLeftFastRebound: number, rearRightFastRebound: number) {
  return rearLeftFastRebound >= this.rearFastReboundMin && rearLeftFastRebound <= this.rearFastReboundMax
    && rearRightFastRebound >= this.rearFastReboundMin && rearRightFastRebound <= this.rearFastReboundMax;
}
}

File: backend/src/main/java/com/simsetuplab/backend/model/setupValidator/ValidateDampers.java
package com.simsetuplab.backend.model.setupValidator;

import com.simsetuplab.backend.model.setup.setupvalues.Dampers;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Builder
@Table(name = "validate_dampers", schema = "public")
public class ValidateDampers {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private int frontBumpMin;
	private int frontBumpMax;
	private int frontFastBumpMin;
	private int frontFastBumpMax;
	private int frontReboundMin;
	private int frontReboundMax;
	private int frontFastReboundMin;
	private int frontFastReboundMax;
	private int rearBumpMin;
	private int rearBumpMax;
	private int rearFastBumpMin;
	private int rearFastBumpMax;
	private int rearReboundMin;
	private int rearReboundMax;
	private int rearFastReboundMin;
	private int rearFastReboundMax;

	@Transient
	List<String> errorList = new ArrayList<>();

	public List<String> validate(Dampers dampers) {
		validateFrontBump(dampers.getFrontLeftBump(), dampers.getFrontRightBump());
		validateFrontFastBump(dampers.getFrontLeftFastBump(), dampers.getFrontRightFastBump());
		validateFrontRebound(dampers.getFrontLeftRebound(), dampers.getFrontRightRebound());
		validateFrontFastRebound(dampers.getFrontLeftFastRebound(), dampers.getFrontRightFastRebound());
		validateRearBump(dampers.getRearLeftBump(), dampers.getRearRightBump());
		validateRearFastBump(dampers.getRearLeftFastBump(), dampers.getRearRightFastBump());
		validateRearRebound(dampers.getRearLeftRebound(), dampers.getRearRightRebound());
		validateRearFastRebound(dampers.getRearLeftFastRebound(), dampers.getRearRightFastRebound());

		return errorList;
	}

	public void validateFrontBump(int frontLeftBump, int frontRightBump) {
		if (!(frontLeftBump >= frontBumpMin && frontLeftBump <= frontBumpMax
				&& frontRightBump >= frontBumpMin && frontRightBump <= frontBumpMax)) {
			errorList.add("front bump");
		}
	}

	public void validateFrontFastBump(int frontLeftFastBump, int frontRightFastBump) {
		if (!(frontLeftFastBump >= frontFastBumpMin && frontLeftFastBump <= frontFastBumpMax
				&& frontRightFastBump >= frontFastBumpMin && frontRightFastBump <= frontFastBumpMax)) {
			errorList.add("front fast bump");
		}
	}

	public void validateFrontRebound(int frontLeftRebound, int frontRightRebound) {
		if (!(frontLeftRebound >= frontReboundMin && frontLeftRebound <= frontReboundMax
				&& frontRightRebound >= frontReboundMin && frontRightRebound <= frontReboundMax)) {
			errorList.add("front rebound");
		}
	}

	public void validateFrontFastRebound(int frontLeftFastRebound, int frontRightFastRebound) {
		if (!(frontLeftFastRebound >= frontFastReboundMin && frontLeftFastRebound <= frontFastReboundMax
				&& frontRightFastRebound >= frontFastReboundMin && frontRightFastRebound <= frontFastReboundMax)) {
			errorList.add("front fast rebound");
		}
	}

	public void validateRearBump(int rearLeftBump, int rearRightBump) {
		if (!(rearLeftBump >= rearBumpMin && rearLeftBump <= rearBumpMax
				&& rearRightBump >= rearBumpMin && rearRightBump <= rearBumpMax)) {
			errorList.add("rear bump");
		}
	}

	public void validateRearFastBump(int rearLeftFastBump, int rearRightFastBump) {
		if (!(rearLeftFastBump >= rearFastBumpMin && rearLeftFastBump <= rearFastBumpMax
				&& rearRightFastBump >= rearFastBumpMin && rearRightFastBump <= rearFastBumpMax)) {
			errorList.add("rear fast bump");
		}
	}

	public void validateRearRebound(int rearLeftRebound, int rearRightRebound) {
		if (!(rearLeftRebound >= rearReboundMin && rearLeftRebound <= rearReboundMax
				&& rearRightRebound >= rearReboundMin && rearRightRebound <= rearReboundMax)) {
			errorList.add("rear rebound");
		}
	}

	public void validateRearFastRebound(int rearLeftFastRebound, int rearRightFastRebound) {
		if (!(rearLeftFastRebound >= rearFastReboundMin && rearLeftFastRebound <= rearFastReboundMax
				&& rearRightFastRebound >= rearFastReboundMin && rearRightFastRebound <= rearFastReboundMax)) {
			errorList.add("rear fast rebound");
		}
	}

}
